cpp/src/arrow/compute/function_internal.h

```diff
@@ -112,9 +81,23 @@ static inline std::string GenericToString(const std::string& value) {
 }
 
 template <typename T>
-static inline enable_if_t<has_enum_traits<T>::value, std::string> GenericToString(
-    const T value) {
-  return EnumTraits<T>::value_name(value);
+static inline std::string GenericToString(const TimeUnit::type value) {
+  switch (value) {
+    case TimeUnit::type::SECOND:
+      return "SECOND";
+    case TimeUnit::type::MILLI:
+      return "MILLI";
+    case TimeUnit::type::MICRO:
+      return "MICRO";
+    case TimeUnit::type::NANO:
+      return "NANO";
```

Comment:
Shouldn't we use lowercase as for other option enums?
(for example, I see you changed "Ascending" to "ascending")